src/system.rs

```diff
@@ -320,6 +320,7 @@ pub enum MemoryType {
     MemoryMappedIOPortSpace,
     PalCode,
     PersistentMemory,
+    Custom(u32),
```

Comment:
This increases `sizeof(MemoryType)` to 8 on 32bit machines, which would violate the spec. Speaking of this, the UEFI specification is quite misleading on this front, anyway. They define the enum but do not specify the type (and the C standard allows multiple possible options). Other sections of the spec explicitly use `u32`, though. So I think we can expect the type to always be an `u32`.

I am unsure what to do about this. I don't think there is a nice way to add "anything-else" options in rust enums, but I might be missing something. The related RFC is:

    https://github.com/rust-lang/rfcs/blob/master/text/2195-really-tagged-unions.md

I have to say I am tempted to mark enums as `u32` explicitly, then also add `Into(u32)` conversions and require function calls and dependent types to take `u32` rather than `MemoryType` as type. That is not particularly nice, but I don't see any better option right now?

Reply:
Ouch. You're right.

I agree that changing `MemoryType` to `u32` in signatures is probably the only way to get this working right now. The spec explicity mentions the `0x8000_0000..0xffff_ffff` range (and no range beyond that) so I think it is a safe bet they actually "mean" `u32`.

I'll close this PR for now.